server/commands/list_objects.go

```diff
@@ -94,13 +157,49 @@ func (q *ListObjectsQuery) Execute(ctx context.Context, req *openfgapb.ListObjec
 	}, nil
 }
 
-func (q *ListObjectsQuery) ExecuteStreamed(ctx context.Context, req *openfgapb.StreamedListObjectsRequest, srv openfgapb.OpenFGAService_StreamedListObjectsServer) error {
+func (q *ListObjectsQuery) ExecuteStreamed(
+	ctx context.Context,
+	req *openfgapb.StreamedListObjectsRequest,
+	srv openfgapb.OpenFGAService_StreamedListObjectsServer,
+) error {
 	err := q.validateInput(ctx, req.StoreId, req.Type, req.AuthorizationModelId, req.Relation)
 	if err != nil {
 		return err
 	}
 
-	resultsChan := make(chan string, q.ListObjectsMaxResults)
+	targetObjectType := req.GetType()
+	targetRelation := req.GetRelation()
+
+	model, err := q.Datastore.ReadAuthorizationModel(ctx, req.GetStoreId(), req.GetAuthorizationModelId())
+	if err != nil {
+		if errors.Is(err, storage.ErrNotFound) {
+			return serverErrors.TypeNotFound(targetObjectType)
+		}
+		return err
+	}
+
+	typesys := typesystem.New(model)
+
+	_, ok := typesys.GetTypeDefinition(targetObjectType)
+	if !ok {
+		return serverErrors.TypeNotFound(targetObjectType)
+	}
+
+	relation, ok := typesys.GetRelation(targetObjectType, targetRelation)
+	if !ok {
+		return serverErrors.RelationNotFound(targetRelation, targetObjectType, nil)
+	}
+
+	hasTypeInfo, ok := typesys.HasTypeInfo(targetObjectType, targetRelation)
+	if !ok {
+		q.Logger.WarnWithContext(ctx, fmt.Sprintf("failed to lookup type info for relation '%s' on object type '%s'", targetRelation, targetObjectType))
+	}
```

Comment:
why is all this code duplicate? i'd put it inside `validateInput`

Reply:
I'm working on refactoring this whole thing. We have duplication across the streamed and regular that doesn't need to exist either. The logic can be shared for many things. I'll let you know when I've updated.

Reply:
Alright, I refactored into a shared `handler` that is used for `ExecuteStreamed` and `Execute`. It's a lot nicer than it was before in v0 and what I had before my most recent commit. Let me know what you think of the newly refactor structure. Way less duplication!